components/Selector/LeadersFilterSelector.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+import {
+  Container,
+  ButtonContent,
+  DropdownButton,
+  DropdownItem,
+  DropdownList,
+  Caret,
+} from './styles';
+
+export type LeadersFilter = 'Skaters' | 'Forwards' | 'Defensemen' | 'Goalies';
+
+interface Props {
+  activeFilter: LeadersFilter;
+  onChange: (type: LeadersFilter) => void;
+}
+
+const FILTERS: {
+  [key: string]: LeadersFilter;
+} = {
+  SKATERS: 'Skaters',
+  FORWARDS: 'Forwards',
+  DEFENSEMEN: 'Defensemen',
+  GOALIES: 'Goalies'
+};
+
+function LeadersFilterSelector({ activeFilter, onChange }: Props): JSX.Element {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [selectedLeadersFilter, setSelectedLeadersFilter] = useState(activeFilter);
+  const selectorRef = useRef(null);
+  const onMouseLeave = () => {
+    setIsExpanded(false);
+    if (selectorRef.current) {
+      selectorRef.current.removeEventListener('mouseleave', onMouseLeave);
+    }
+  };
+
+  useEffect(() => {
+    if (isExpanded && selectorRef.current) {
+      selectorRef.current.addEventListener('mouseleave', onMouseLeave);
+    }
+  }, [selectorRef, isExpanded]);
+
+  useEffect(() => setSelectedLeadersFilter(activeFilter), [activeFilter]);
+
+  const onButtonClick = () => setIsExpanded(!isExpanded);
```

Comment:
```suggestion
  const onButtonClick = useCallback(() => setIsExpanded((expanded) => !expanded), [setIsExpanded]);
```
Let's start trying to incorporate some more `useCallbacks` here. Make sure that this works properly first though

Reply:
I've made changes to use `useCallback` where suggested. Tested them as well and everything still works.

I'll keep in mind to start using this consistently from now on.

Reply:
Awesome! Thanks! 

These usually are helpful with handlers and will reduce re-renders